src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -1347,5 +1317,200 @@ private void VerifyNotepadMainWindowTitle(Process process, string filename)
+            {
+                LoadUserProfile = true,
+                UseCredentialsForNetworkingOnly = true,
+                UserName = "dummy",
+                PasswordInClearText = "not used, because ArgumentException should be thrown before",
+                FileName = "notepad.exe",
+                Arguments = null,
+                WindowStyle = ProcessWindowStyle.Minimized
+            };
+
+            Assert.Throws<ArgumentException>("startInfo", () =>
+            {
+                using (var process = Process.Start(info))
+                {
+                    Assert.False(process != null, $"Process started despite incompatible options {nameof(info.LoadUserProfile)} and {nameof(info.UseCredentialsForNetworkingOnly)} were enabled");
+                }
+            });
+        }
+
+        [ConditionalFact(nameof(IsAdmin_IsNotNano_RemoteExecutorIsSupported))] // Nano has no "netapi32.dll", Admin rights are required
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [OuterLoop("Requires admin privileges")]
+        public void TestUserNetworkCredentialsPropertiesOnWindows()
+        {
+            string testFilePath = GetTestFilePath();
+            Assert.False(string.IsNullOrWhiteSpace(testFilePath), $"Path to test file should not be empty: {testFilePath}");
+
+            string testFilePathRoot = Path.GetPathRoot(testFilePath);
+            const string ShareName = "testForDotNet";
+            using TestFileShare fileShare = new TestFileShare(ShareName, Path.GetDirectoryName(testFilePath));
+            string testFileUncPath = $"\\\\localhost\\{ShareName}\\{Path.GetFileName(testFilePath)}";
+            string testFileContent = "42";
+            File.WriteAllText(testFilePath, testFileContent);
+
+            const string LocalPath = nameof(LocalPath);
+            const string UncPath = nameof(UncPath);
+            using Process p = CreateProcess(() =>
+            {
+                // Read file content using network credentials and output it for assertion comparison
+                _ = File.ReadAllText(Environment.GetEnvironmentVariable(UncPath));
+                try
+                {
+                    // Read file content using current user - should fail, because impersonated user
+                    // isn't explicitly authorized to access the file.
+                    _ = File.ReadAllText(Environment.GetEnvironmentVariable(LocalPath));
+
+                    return -1;
+                }
+                catch (Exception ex) when (ex is SecurityException or UnauthorizedAccessException)
+                {
+                    return RemoteExecutor.SuccessExitCode;
+                }
+            });
+            p.StartInfo.Environment[LocalPath] = testFilePath;
+            p.StartInfo.Environment[UncPath] = testFileUncPath;
+            p.StartInfo.UseCredentialsForNetworkingOnly = true;
+
+            using var processInfo = CreateUserAndExecute(p, true, Setup, Cleanup);
```

Comment:
I'm getting `"Access is denied."` errors when trying to impersonate. It doesn't *appear* to be related to permissions, because it even happens if I don't use the `RemotExecutor`, but just use a simple process that starts `C:\Windows\System32\notepad.exe`. Maybe there's some minor thing I'm missing?

Basically, this is stopping me from verifying I'm not accidentally reading the file as the current admin user, which would obviously succeed, making the test that the "netonly" user can read the file void.

Reply:
> I am not sure if I understand correctly, but your account might not have the permissions to impersonate.

Might be possible, even though I ran the tests as a local admin in an elevated `cmd`/`pwsh`. I was also able to impersonate in `cmd` using `runas`. I don't know if there's a special permission for impersonation, but will have a look. Thanks!

> How about running as admin and extending the tests to verify current user name?

The current user of the `CreateProcess` process probably doesn't change if we use `UseCredentialsForNetworkingOnly`, so it can't be detected. https://stackoverflow.com/a/38390794/11963

Reply:
> verifying I'm not accidentally reading the file as the current admin user, which would obviously succeed,

I haven't followed this discussion, so this may be random, but an admin can of course modify perms to deny access to itself - if that helps here.

Reply:
> > verifying I'm not accidentally reading the file as the current admin user, which would obviously succeed,
> 
> I haven't followed this discussion, so this may be random, but an admin can of course modify perms to deny access to itself - if that helps here.

Definitely! I tried that, too, but it failed for some reason I can't remember, so I ditched the approach at first. I think it's the one of the least intrusive ways to write this test so, and so think I'll have another look at this, too. Thanks for jumping in, @danmoseley!

Reply:
https://github.com/dotnet/runtime/blob/cb281adedfef0ab34aba947320385f3999f9f82f/src/libraries/System.IO.FileSystem/tests/Directory/Delete.Windows.cs#L23

Reply:
Thanks! Will test that in the coming days. (It's getting late :wink:)